### DISCUSSION TITLE
[ 命名建議 ] LeetCode 496. Next Greater Element I

### DISCUSSION THREAD

**Discussion by LPenny-github** [EngTW/English-for-Programmers#117] [Q&A] [answered]
LeetCode:
https://leetcode.com/problems/next-greater-element-i/

線上教材:
https://blog.techbridge.cc/2019/10/26/leetcode-pattern-next-greater-element/

---

nums1 和 nums2 我猶豫很久，還是不知如何命名，因此沒有為他們改名 orz

再麻煩 @twy30 給予命名建議 orz 感激不盡

```csharp
using System.Collections.Generic;

public class Solution
{
    public int[] NextGreaterElement(int[] nums1, int[] nums2)
    {
        Stack<int> nextGreaterNumber = new Stack<int>();
        Dictionary<int, int> validPair = new Dictionary<int, int>();

        for (int i = 0; i < nums2.Length; ++i)
        {
            while (nextGreaterNumber.Count > 0 && nextGreaterNumber.Peek() < nums2[i])
            {
                validPair.Add(nextGreaterNumber.Pop(), nums2[i]);
            }
            nextGreaterNumber.Push(nums2[i]);
        }

        int[] output = new int[nums1.Length];

        for (int j = 0; j < nums1.Length; ++j)
        {
            output[j] = validPair.ContainsKey(nums1[j])? validPair[nums1[j]]: -1;
        }

        return output;
    }
}
```

**Comment by twy30** [comment, marked as answer]
@LPenny-github 

你好 😊

---

> nums1 和 nums2 我猶豫很久，還是不知如何命名，因此沒有為他們改名 orz

能理解 😅

可以考慮：

* `nums1`: `inputNumbers`
  * 因為最後 `output` 是由「一一處理此 array 的成員」而來，所以對應到 `inputNumbers`
* `nums2`: `nextGreaterNumberCandidates`
  * 「 `nextGreaterNumber` 的候選者」

---

```C#
        Stack<int> nextGreaterNumber = new Stack<int>();
```

可以考慮：

* `nextGreaterNumberStack`
* `nextGreaterNumberCandidateStack`

---

```C#
        Dictionary<int, int> validPair = new Dictionary<int, int>();
```

可以考慮：

* `nextGreaterNumbers`
* `nextGreaterNumberTable`
* `nextGreaterNumberLookupTable`

**Reply by LPenny-github** [reply]
@twy30 感謝大大 orz
